akka-stream/src/main/scala/akka/stream/impl/PhasedFusingActorMaterializer.scala

```diff
@@ -736,8 +738,14 @@ private final case class SavedIslandData(islandGlobalOffset: Int, lastVisitedOff
         fuseIntoExistingInterperter(shell)
 
       case _ ⇒
+
+        val dispatcher =
+          effectiveAttributes.mandatoryAttribute[ActorAttributes.Dispatcher] match {
+            case ActorAttributes.IODispatcher           ⇒ settings.blockingIoDispatcher
+            case ActorAttributes.Dispatcher(dispatcher) ⇒ dispatcher
+          }
```

Comment:
This code is now duplicated across quite a few places. Also the fact that `IODispatcher` has a `dispatcher` attribute and it is not used here makes it a bit confusing.

Reply:
Added an issue to track that #24604